streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -251,26 +252,69 @@ public StreamTask(final TaskId id,
         if (eosEnabled) {
             initializeTransactions();
         }
-    }
 
-    @Override
-    public boolean initializeStateStores() {
-        log.debug("Initializing state stores");
+        try {
+            final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata = consumer.committed(partitions);
+            initializeCommittedOffsets(offsetsAndMetadata);
+            initializeTaskTime(offsetsAndMetadata);
+        } catch (final AuthorizationException e) {
+            throw new ProcessorStateException(String.format("task [%s] AuthorizationException when initializing offsets for %s", id, partitions), e);
+        } catch (final WakeupException e) {
+            throw e;
+        } catch (final KafkaException e) {
+            throw new ProcessorStateException(String.format("task [%s] Failed to initialize offsets for %s", id, partitions), e);
+        }
+    }
 
+    private void initializeCommittedOffsets(final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata) {
```

Comment:
This is a mix of "old" `initializeStateStores` (the part to filter for source topic partitions that are use as changelogs) and "old" `AbstractTaks#committedOffsetForPartitions`)